### PR TITLE
issue: 3325307 CI OS list updated

### PR DESCRIPTION
**Pull request by haron**
Signed-off-by: Alexander Ryabov <aryabov@nvidia.com>

## Description
Added build in RHEL 8.6 and 9.0 containers, obsolete OS versions removed.

## Change type
What kind of change does this PR introduce?
- [x] Build related changes
- [x] CI related changes

### CODE REVIEW

File: .ci/matrix_job.yaml [hunk 1]
```diff
@@ -27,21 +27,14 @@ env:
...
-#  - {name: 'ub20.04-mofed-ppc64le', url: 'harbor.mellanox.com/swx-infra/ppc64le/ubuntu20.04/builder:mofed-5.2-1.0.4.0', category: 'base', arch: 'ppc64le'}
 # inbox
-  - {name: 'ub21.10-inbox-x86_64',  url: 'harbor.mellanox.com/hpcx/x86_64/ubuntu21.10/builder:inbox', category: 'base', arch: 'x86_64'}
-  - {name: 'rhel8.3-inbox-x86_64',  url: 'harbor.mellanox.com/hpcx/x86_64/rhel8.3/builder:inbox', category: 'base', arch: 'x86_64'}
-  - {name: 'rhel8.3-inbox-aarch64', url: 'harbor.mellanox.com/hpcx/aarch64/rhel8.3/builder:inbox', category: 'base', arch: 'aarch64'}
-# legacy
-  - {name: 'rhel7.6-mofed-x86_64', url: 'harbor.mellanox.com/hpcx/x86_64/rhel7.6/builder:mofed-5.1-0.6.6.0', category: 'base', arch: 'x86_64'}
-  - {name: 'rhel7.6-inbox-x86_64', url: 'harbor.mellanox.com/hpcx/x86_64/rhel7.6/builder:inbox', category: 'base', arch: 'x86_64'}
+  - {name: 'rhel8.6-inbox-x86_64',  url: 'harbor.mellanox.com/hpcx/x86_64/rhel8.6/builder:inbox', category: 'base', arch: 'x86_64'}
+  - {name: 'rhel8.6-inbox-aarch64', url: 'harbor.mellanox.com/hpcx/aarch64/rhel8.6/builder:inbox', category: 'base', arch: 'aarch64'}
```
Comment (AlexanderGrissik) [inline .ci/matrix_job.yaml:37]: we need ubuntu 20.04 aarch mofed instead

File: .ci/matrix_job.yaml [hunk 2]
```diff
@@ -27,21 +27,14 @@ env:
 
 runs_on_dockers:
 # mofed
-  - {name: 'rhel8.3-mofed-x86_64', url: 'harbor.mellanox.com/hpcx/x86_64/rhel8.3/builder:mofed-5.2-0.5.3.0', category: 'base', arch: 'x86_64'}
-  - {name: 'sl15sp2-mofed-x86_64', url: 'harbor.mellanox.com/swx-infra/x86_64/sles15sp2/builder:mofed-5.2-2.2.0.0', category: 'base', arch: 'x86_64'}
-  - {name: 'fc31-mofed-x86_64',    url: 'harbor.mellanox.com/swx-infra/x86_64/fedora31/builder:mofed-5.1-1.0.7.0', category: 'base', arch: 'x86_64'}
```
[the remark this answers is not in the document]
Comment (AlexanderGrissik) [reply .ci/matrix_job.yaml:32]: fedora 36

**Comment by haron** [comment]
bot:retest

**Comment by lennybe** [comment]
bot:retest

**Comment by haron** [comment]
bot:retest

**Review by AlexanderGrissik — APPROVED** [review]

**Pull request merged**